### ISSUE TITLE
Filter and Columns Localization by switching language is not working

### ISSUE THREAD

**Issue by kalbunna** [bogdan/datagrid#169]
I have use Datagrid to filter and show data . Now I try to apply multiple languages but the header always show the first configuration language. For instance, let say that I have two locale files
 en.yml

```
datagrid:
    form:
      search: Search
      reset: Reset
    columns:
      clients:
        name: Name

```
and km.yml

 ```
datagrid:
    form:
      search: ស្វែងរក
      reset: សំអាត
    columns:
      clients:
        name: អាត
```

In the filter i just call : 
`filter(:name, :string, header: I18n.t('datagrid.columns.clients.name')}) { |value, scope| scope.first_name_like(value) }`

I am waiting to hearing from you.
Thank you in advance 

Kind regards

**Comment by bogdan** [comment]
Datagrid uses rails locale setup - whatever locale is used by rails - will be used by datagrid.

How did you change the locale?
What is the locale at `I18n.locale` ? 

Try:

``` ruby
I18n.with_locale(:km) { I18n.t('datagrid.columns.clients.name')}
```

What is the result?

**Comment by kalbunna** [comment]
Thank for your quick reply.
I have change default locale in application.rb to: 
`config.i18n.default_locale = :en`
and in application_controller.rb I have called:
` before_action: set_locale`
```
def set_locale
    I18n.locale = params[:locale] || I18n.default_locale
  end

  def default_url_options(options={})
    { locale: I18n.locale }.merge options
  end
```
And when I try  in console:
`I18n.with_locale(:km) { I18n.t('datagrid.columns.clients.name')}`
It shown me the text in km language but it should dynamically switching I18n.with_locale(:km) and switching I18n.with_locale(:en). There is a way to make it dynamic change everytime I switch language?

Thank you!

**Comment by bogdan** [comment]
``` ruby
around_filter :apply_locale

def apply_locale
  I18n.with_locale(params[:locale] || I18n.default_locale) { yield }
end
```

**Comment by kalbunna** [comment]
I have placed this above code into applicaiton_controller.rb. Am I right?

and in the filter I just call something like

`filter(:first_name, :string, header: I18n.t('datagrid.columns.clients.name')) { |value, scope| scope.first_name_like(value) }`

I am newbie for Datagrid, hope you can help me by give me more details on working on it. 

Thank!

**Comment by bogdan** [comment]
The only advice I can give now is make sure your localizations are working without datagrid. 90% chance that the problem is not related to it at all.

**Comment by kalbunna** [comment]
Everything without datagrid are working fine include Search and Reset button. I afraid that the problem is due to inline header property is not reload again after I switch to new language. 

Thank !

**Comment by bogdan** [comment]
Hm, this is a true bug... I'll fix it in next 24 hours.

**Comment by bogdan** [comment]
Released 1.4.2 to fix an issue.

You need to wrap a header with proc in order to make it dynamic because it will be executed only once by ruby otherwise.


`filter(:name, type, header: -> {I18n.t('datagrid.columns.clients.name')})`

**Comment by kalbunna** [comment]
You are genius man, thank you very much for your kind help and distributed kind of good gem for other developer to use. 

Many thank !
